test/model/LocationsSpec.scala

```diff
@@ -16,87 +16,58 @@
+import org.scalatest.mock.MockitoSugar
 import uk.gov.hmrc.play.test.UnitSpec
 
-class LocationsSpec extends UnitSpec {
+class LocationsSpec extends UnitSpec with MockitoSugar {
 
-  "Locations" should {
-    "return a Location from the configuration" in new Setup {
+  "locations" should {
 
-      implicit val fakeApplication = FakeApplication(additionalConfiguration = additionalConfiguration)
+    "return Location when config has no missing keys" in {
+      val mockAppConfig = mock[AppConfig]
+      when(mockAppConfig.getLocationConfig("pta", "name")).thenReturn(Some("some-name"))
+      when(mockAppConfig.getLocationConfig("pta", "url")).thenReturn(Some("some-url"))
 
-      running(fakeApplication) {
-        val locations = new Locations {}
-
-        val location = locations.locationFromConf("some-location")
-
-        val expectedLocation = Location(locationName, locationUrl, Map("continue" -> continueQueryParam, "failure" -> failureQueryParam, "origin" -> originQueryParam))
-
-        location shouldBe expectedLocation
+      val locations = new Locations {
+        val appConfig = mockAppConfig
       }
-    }
 
-    "throw RuntimeException when expected location configuration element is not existing" in {
+      val result = locations.PersonalTaxAccount
 
-      implicit val application = FakeApplication()
-
-      running(application) {
-        val locations = new Locations {}
-
-        val thrown = intercept[RuntimeException] {
-          locations.locationFromConf("some-location")
-        }
-
-        thrown.getMessage shouldBe "location configuration not defined for some-location"
-      }
+      result shouldBe Location("some-name", "some-url")
     }
 
-    "throw RuntimeException when location name is missing from configuration" in new Setup {
+    "return Location when config has no missing url key" in {
+      val mockAppConfig = mock[AppConfig]
+      when(mockAppConfig.getLocationConfig("tax-account-router", "name")).thenReturn(Some("some-name"))
+      when(mockAppConfig.getLocationConfig("tax-account-router", "url")).thenReturn(None)
 
-      implicit val fakeApplication = FakeApplication(additionalConfiguration = additionalConfiguration - "locations.some-location.name")
-
-      running(fakeApplication) {
-        val locations = new Locations {}
-
-        val thrown = intercept[RuntimeException] {
-          locations.locationFromConf("some-location")
-        }
-
-        thrown.getMessage shouldBe "name not configured for location - some-location"
+      val locations = new Locations {
+        val appConfig = mockAppConfig
       }
-    }
 
-    "throw RuntimeException when location url is missing from configuration" in new Setup {
+      val caught = intercept[RuntimeException] {
+        locations.TaxAccountRouterHome
+      }
 
-      implicit val fakeApplication = FakeApplication(additionalConfiguration = additionalConfiguration - "locations.some-location.url")
+      caught shouldNot be(null)
```

Comment:
should you make any further assertions on the caught exception? to make sure that it's the one you're expecting